Color Schemes: update sidebar styles to better match wp-admin

chad1008 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Part of 7884-gh-Automattic/dotcom-forge

## Proposed Changes

Adds some styles and a few new variables to the Calypso sidebar stylesheet to allow more versatility within individual color schemes to better match wp-admin

This PR contains the changes of 17 other smaller PRs for the individual color schemes (see links in the to-do list on 7884-gh-Automattic/dotcom-forge).

## Why are these changes being made?

An [issue](https://github.com/Automattic/wp-calypso/pull/92005) recently highlighted some disconnects that have developed between our unified/redesigned sidebars and the wp-admin styles for various color schemes. This PR is the first step in making sure our schemes can match what wp-admin displays.

### Important Note On Merging:

This PR is a pre-requisite for other PRs for each individual color scheme. I wanted to keep them separate so an individual scheme is easy to revert if needed.

That means this PR has to be merged before any of the others listed in 7884-gh-Automattic/dotcom-forge. It also means, because this PR introduces a few variables that the color schemes on production don't recognize, that we should merge this one and then immediately merge the individual scheme PRs. Ideally, I'd like to merge them all in one go so they can all stack into the same deploy so there's no lag time where specific sidebar elements will be unstyled. The new variables (see below) for the hover state of currently selected submenu items would be the most impactful one if there was a delay on a given scheme.

### New variables:

- `--color-sidebar-submenu-selected-hover-text`: On some schemes, the currently selected submenu item (e.g. **Posts > All Posts**) does not change its color when hovered. On other schemes, the hover state matches that of unselected submenu items and the color does change. This variable allows us to make that determination on a per-scheme basis.
- `--color-navredesign-sidebar-submenu-selected-hover-text`: This variable accomplishes the same thing as the previous one, but for the Nav Redesign variant of the sidebar, which appears when a site has `wp-admin` set in interface setting and the My Home/Hosting page is being viewed (this is the only Calypso page that gets rendered when a `wp-admin` preference is set.
- `--color-collapse-menu-text`: Different themes handle the color of this menu differently. Sometimes it matches the base sidebar text color, other times the submenu hover color, or maybe just a separate color all together. The hover state seems to be consistent in that it matches the submenu hover color across all schemes, so we thankfully don't need additional specificity there.


## Testing Instructions

This PR alone won't fully address most schemes, so I'd recommend visual code review to confirm the new variables and changes to existing styles all make sense. When reviewing the schemes themselves the styles will all come together, making a review of the UI more practical.

The individual scheme PRs are all branched off of this one, and can be rebased if we make any changes here to keep them current.

chad1008 committed:
fix selected menu item

chad1008 committed:
add specifity to sidebar styles

chad1008 committed:
add specifity to sidebar styles

chad1008 committed:
fix expandable and selected background color

chad1008 committed:
fix expandable and selected background color

chad1008 committed:
add variable to target the "collapse menu" link

chad1008 committed:
add variable to target the "collapse menu" link

chad1008 committed:
fix "collapse menu" link color

chad1008 committed:
fix submenu selected hover color

chad1008 committed:
fix "collapse menu" color

chad1008 committed:
add var for submenu selected hover color

chad1008 committed:
add var for submenu selected hover color

chad1008 committed:
fix navredesign submenu hover color

chad1008 committed:
fix submenu selected hover color

chad1008 committed:
fix "collapse menu" link color

chad1008 committed:
fix "collapse menu" link color

chad1008 committed:
fix navredesign submenu hover color

chad1008 committed:
fix submenu selected hover color

chad1008 committed:
fix submenu hover colors

chad1008 committed:
fix submenu colors

chad1008 committed:
fix submenu selected hover color

chad1008 committed:
add navredesign submenu selected hover variable

chad1008 committed:
add navredesign submenu selected hover variable

chad1008 committed:
fix submenu and "collapse menu" colors

chad1008 committed:
fix subment selected hover color

chad1008 committed:
fix nav redesign submenu and "collapse menu" colors

chad1008 committed:
fix "collapse menu", nav redesign and regular sidebar submenu selected colors

chad1008 committed:
fix submenu text color

chad1008 committed:
fix submenu selected hover color and "collapse menu" color

chad1008 committed:
fix nav redesign colors and "collapse menu" color

chad1008 committed:
fix submenu selected hover color

chad1008 committed:
fix navredesign colors and "collapse menu" button

chad1008 committed:
fix submenu hover colors

chad1008 committed:
reorganize existing style

nightnei reviewed (approved):
LGTM 👍

@chad1008 
>Ideally, I'd like to merge them all in one go so they can all stack into the same deploy so there's no lag time where specific sidebar elements will be unstyled.

Hm, in this case I didn't get the point to split them into different PRs 😅

chad1008 commented:
Thanks, Vova!

> Hm, in this case I didn't get the point to split them into different PRs 😅

Yeah, I went back and forth on it. Ultimately I wanted to make sure an individual scheme was easy to revert indivdually if needed. I don't think that's likely, but better safe than sorry. I also thought testing one massive PR across two different Calypso sidebars and 15ish schemes would be a nightmare. 😆

chad1008 committed:
Update packages/calypso-color-schemes/src/shared/color-schemes/_aquatic.scss

Co-authored-by: Volodymyr Makukha <nei.css@gmail.com>

chad1008 committed:
edit code comment

chad1008 committed:
edit code comment

chad1008 committed:
edit code comment

chad1008 committed:
edit code comment

chad1008 committed:
edit code comment

chad1008 committed:
edit code comment

chad1008 committed:
edit code comment

chad1008 committed:
edit code comment

chad1008 committed:
edit code comment

chad1008 committed:
edit code comment

chad1008 committed:
edit code comment

chad1008 committed:
reorganize code

chad1008 committed:
remove empty section

chad1008 committed:
remove empty section

chad1008 committed:
remove empty section

chad1008 committed:
remove empty section

chad1008 committed:
remove empty line

chad1008 committed:
remove empty section

chad1008 committed:
reorganize code

chad1008 committed:
merging modern color scheme updates to main PR

Merge branch 'chad1008/fix-modern-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
merging aquatic color scheme updates to main PR

Merge branch 'chad1008/fix-aquatic-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
merging blue  color scheme updates to main PR

Merge branch 'chad1008/fix-blue-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
merging classic blue color scheme updates to main PR

Merge branch 'chad1008/fix-classic-blue-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
merging classic bright color scheme updates to main PR

Merge branch 'chad1008/fix-classic-bright-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
merging classic-dark color scheme updates to main PR

Merge branch 'chad1008/fix-classic-dark-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
merging coffee color scheme updates to main PR

Merge branch 'chad1008/fix-coffee-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
merging contrast color scheme updates to main PR

Merge branch 'chad1008/fix-contrast-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
merging ectoplasm color scheme updates to main PR

Merge branch 'chad1008/fix-ectoplasm-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
merging midnight color scheme updates to main PR

Merge branch 'chad1008/fix-midnight-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
merging nightfall color scheme updates to main PR

Merge branch 'chad1008/fix-nightfall-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
Merge branch 'chad1008/fix-ocean-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
Merge branch 'chad1008/fix-powder-snow-color-scheme' into chad1008/sidebar-style-updates

chad1008 committed:
Merge branch 'chad1008/fix-sakura-color-scheme' into chad1008/sidebar-style-updates
